probe-rs/src/probe/daplink/tools.rs

```diff
@@ -180,10 +180,11 @@ pub fn open_device_from_selector(
             };
 
             let timeout = Duration::from_millis(100);
-            let language = handle.read_languages(timeout)?[0];
             let sn_str = handle
-                .read_serial_number_string(language, &d_desc, timeout)
-                .ok();
+                .read_languages(timeout)?
+                .get(0)
+                .map(|lang| handle.read_serial_number_string(*lang, &d_desc, timeout))
+                .transpose()?;
```

Comment:
Thanks for fixing this!

There is still a problem with this tho: If the device has no SerialNo, which is perfectly valid, the ::transpose() will actually return a Err() which then will return early again. But having no serial number really isn't a dealbreaker. So this really needs to be moved towards where the serial number really is checked (after VID/PID validation) :)